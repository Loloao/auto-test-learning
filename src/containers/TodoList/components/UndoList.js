import React, { Component } from 'react'

class UndoList extends Component {
  render() {
    const { list, deleteItem } = this.props
    return (
      <div>
        <div data-test="count">{list.length}</div>
        <ul>
          {list.map((v, i) => {
            return (
              <li data-test="list-item" key={`${v}-${i}`}>
                {v}
                <span data-test="delete-item" onClick={deleteItem(i)}>
                  -
                </span>
              </li>
            )
          })}
        </ul>
      </div>
    )
  }
}

export default UndoList
