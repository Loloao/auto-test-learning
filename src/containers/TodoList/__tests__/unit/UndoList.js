import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import UndoList from '../../components/UndoList'
import { findTestWrapper } from '../../../../utils/testUtils'
import Adapter from 'enzyme-adapter-react-16'
Enzyme.configure({ adapter: new Adapter() })

it('未完成列表初始化 count 数目为 0，列表为无内容', () => {
  const wrapper = shallow(<UndoList list={[]} />)
  const countElem = findTestWrapper(wrapper, 'count')
  const listItems = findTestWrapper(wrapper, 'list-item')
  expect(countElem.text()).toEqual('0')
  expect(listItems.length).toBe(0)
})

it('未完成列表当数据有内容时 count 数目显示数据长度，列表不为空', () => {
  const listData = ['学习Jest', '学习TDD', '学习单元测试']
  const fn = jest.fn()
  const wrapper = shallow(<UndoList list={listData} deleteItem={fn} />)
  const countElem = findTestWrapper(wrapper, 'count')
  const listItems = findTestWrapper(wrapper, 'list-item')
  expect(countElem.text()).toEqual('3')
  expect(listItems.length).toBe(3)
})

it('未完成列表当数据有内容时，要存在删除按钮', () => {
  const listData = ['学习Jest', '学习TDD', '学习单元测试']
  const fn = jest.fn()
  const wrapper = shallow(<UndoList list={listData} deleteItem={fn} />)
  const deleteItems = findTestWrapper(wrapper, 'delete-item')
  expect(deleteItems.length).toBe(3)
})
