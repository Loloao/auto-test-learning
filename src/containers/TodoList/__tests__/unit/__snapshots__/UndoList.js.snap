// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 组件渲染样式正常 1`] = `
<div
  className="header"
>
  <div
    className="header-content"
  >
    TodoList
    <input
      className="header-input"
      data-test="input"
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder="Todo"
      value=""
    />
  </div>
</div>
`;

exports[`Un 组件渲染样式正常 1`] = `
<div>
  UndoList
</div>
`;

exports[`未完成列表初始化 count 数目为 0 1`] = `
<div>
  UndoList
</div>
`;

exports[`未完成列表初始化 count 数目为 0，列表为无内容 1`] = `
<div>
  UndoList
</div>
`;
